### ISSUE TITLE
#262 follow-up — fix Stage A0 initialization fingerprint and decide fresh-execution admissibility

### ISSUE THREAD

**Issue by lisbun** [lisbun/lisjong-arena#307]
Refs #255
Refs #259
Refs #262 — COMPLETE / STOP / INVALID
Refs PR #306

## Purpose

Resolve the execution-infrastructure defect that invalidated #262 before any scientific Gate result was produced, and make an explicit protocol decision on whether a fresh Stage A0 execution is admissible.

This Issue is **not** permission to silently rerun #262.

## Observed #262 terminal result

```text
PREFLIGHT PASS
TRAIN rows = 9116
VALIDATION rows = 2555

STOP / INVALID
stage = E1 A/T training
error = A/T initial Policy state fingerprint drifted

result_identity =
760c7e81a4dfe47d886c55755319f2b5304d8b6bdec1eefb9452126e2c68495e

protected_test_evaluated = false
```

No Gate A0.5 result was produced.
No downstream game was run.

## Suspected implementation defect

The merged #262 executor defines the initialization fingerprint by serializing a `state_dict` through `torch.save(..., BytesIO())` and hashing the resulting bytes.

It then recreates a model, strict-loads cloned tensors, serializes the new `state_dict` again, and requires the serialization SHA-256 to match.

That is stronger than the scientific requirement and is not a safe canonical tensor-content equality contract. Equal tensor values can be represented by different serialization byte streams.

The scientific contract only requires, within each seed:

```text
A_s initial shared encoder bytes/content
==
T_s initial shared encoder bytes/content

A_s initial Policy-head bytes/content
==
T_s initial Policy-head bytes/content
```

The implementation must verify actual parameter/buffer content, not incidental `torch.save` container bytes.

## Phase R0 — reproduce and prove the defect

Under the #262 locked ML runtime (at minimum PyTorch 2.13.0, Python 3.14.6), add a focused test that:

1. creates the locked Policy model under a fixed seed;
2. clones its `state_dict`;
3. strict-loads that state into a fresh model;
4. proves every key, dtype, shape, and tensor value is identical;
5. demonstrates whether the current serialization-byte fingerprint can differ.

Do not read scientific TRAIN/VALIDATION artifacts for this reproduction.

## Phase R1 — define a canonical initialization identity

Replace the `torch.save`-byte fingerprint with a deterministic tensor-content fingerprint.

The fingerprint must bind, in deterministic key order, at least:

```text
state key
dtype
shape
raw contiguous CPU tensor bytes
```

Buffers, if any, must be covered exactly like parameters.

The same function must produce the same identity:
- before and after strict `load_state_dict`;
- for A and T initial Policy states within one seed;
- across repeated calls in the same locked runtime.

Also retain an explicit elementwise equality assertion for A/T initialization rather than relying only on the digest.

## Phase R2 — regression coverage

Add ML tests that execute the real initialization path, including `train_seed_pair()` far enough to cross the initialization check.

Coverage must fail under the defective implementation and pass under the fix.

Do not weaken unrelated checkpoint SHA-256 integrity checks: checkpoint file digests may continue to hash exact on-disk bytes. This Issue concerns only the semantic identity of initial tensor state.

## Phase R3 — scientific rerun admissibility decision

Before any fresh scientific execution, explicitly record whether rerun is admissible.

Facts to bind:

```text
#262 terminal result = STOP / INVALID
Gate A0.5 result exposed = false
downstream result exposed = false
protected TEST evaluated = false
failure occurred before optimizer training began
failure arose from execution machinery, not scientific outcome
```

If those facts are verified and the project chooses to authorize a fresh execution, create a **new execution Issue** bound to:
- the same final #259 Lock B scientific protocol unless an explicit reason requires a new lock;
- the corrected Arena execution revision;
- a new output root;
- no reuse/deletion/overwrite of the #262 result artifact.

Do not reopen or rerun #262.

## Acceptance criteria

- [ ] Root cause is reproduced without scientific data.
- [ ] Canonical tensor-content fingerprint is implemented.
- [ ] A/T initial Policy equality is checked directly by key/dtype/shape/value.
- [ ] Real initialization path has regression coverage.
- [ ] Existing checkpoint file-integrity SHA checks are not weakened.
- [ ] #262 artifact remains immutable and preserved.
- [ ] No protected TEST is read.
- [ ] No scientific rerun occurs in this Issue.
- [ ] Rerun admissibility is recorded explicitly.
- [ ] If admissible, a separate fresh execution Issue is created.

## Non-goals

Do not change:
- Tenpai target or eligibility;
- Baseline 0 / Baseline 1;
- training seeds;
- optimizer / lambda / class treatment;
- checkpoint selection rule;
- Gate metric / interval / PASS rule;
- downstream opponent population, seeds, rotations, or game budget;
- protected TEST policy.

**Comment by lisbun** [comment]
## R3 decision — fresh execution is admissible

Root cause is now confirmed and fixed by PR #308, merged as Arena revision `3a995e89deff3e89eff9e755b4d1f323e96c12e8`.

Evidence:
- the old initialization fingerprint hashed `torch.save(state_dict)` container bytes;
- the regression test proves independently materialized strict-loaded states can be key/dtype/shape/value-identical while those serialization hashes differ;
- the corrected fingerprint binds deterministic key order + dtype + shape + raw contiguous CPU tensor bytes;
- A/T initial Policy states are now directly checked by key/dtype/shape/`torch.equal`;
- checkpoint on-disk SHA-256 integrity checks are unchanged;
- CI is green for both `quality` and `ml-tests`.

#262 facts:
- terminal result = `STOP / INVALID`;
- failure occurred at E1 initialization verification before optimizer training began;
- Gate A0.5 result exposed = false;
- downstream result exposed = false;
- protected TEST evaluated = false;
- the #262 result artifact remains immutable and must not be deleted or overwritten.

Decision:

```text
FRESH EXECUTION ADMISSIBLE
```

Rationale: no scientific outcome was observed, and the correction changes only execution-infrastructure identity checking, not the #259 scientific protocol. A fresh execution may therefore reuse the exact final #259 Lock B, but it must be performed in a new Issue, from the corrected Arena revision, with a new output root. #262 must not be reopened or rerun.
